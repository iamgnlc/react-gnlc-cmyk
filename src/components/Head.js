import React from "react"
import Helmet from "react-helmet"

import preview from "../assets/images/preview.png"
import favicon from "../assets/favicon.ico"

const Head = React.memo(() => {
  return (
    <Helmet>
      <title>
        {`${process.env.RAZZLE_AUTHOR} / ${process.env.RAZZLE_NAME}`}
      </title>
      <link rel="shortcut icon" href={favicon} />
      <meta name="robots" content="noindex" />
      <meta http-equiv="X-UA-Compatible" content="IE=edge" />
      <meta charset="utf-8" />
      <meta
        name="viewport"
        content="width=device-width, user-scalable=no, initial-scale=1, minimum-scale=1, maximum-scale=1, minimal-ui"
      />
      <meta
        name="title"
        content={`${process.env.RAZZLE_AUTHOR} / ${process.env.RAZZLE_NAME}`}
      />
      <meta name="description" content={process.env.RAZZLE_NAME} />
      <meta name="author" content={process.env.RAZZLE_AUTHOR} />

      <meta
        property="og:site_name"
        content={`${process.env.RAZZLE_AUTHOR} / ${process.env.RAZZLE_NAME}`}
      />
      <meta
        property="og:title"
        content={`${process.env.RAZZLE_AUTHOR} / ${process.env.RAZZLE_NAME}`}
      />
      <meta property="og:description" content={process.env.RAZZLE_NAME} />
      <meta property="og:image" content={preview} />

      <meta
        name="twitter:title"
        content={`${process.env.RAZZLE_AUTHOR} / ${process.env.RAZZLE_NAME}`}
      />
      <meta name="twitter:description" content={process.env.RAZZLE_NAME} />
      <meta name="twitter:image" content={preview} />
      <meta name="twitter:card" content="summary_large_image" />
      <meta name="twitter:site" content={`@${process.env.RAZZLE_AUTHOR}`} />
      <meta
        name="twitter:image:alt"
        content={`${process.env.RAZZLE_AUTHOR} / ${process.env.RAZZLE_NAME}`}
      />

      <meta name="mobile-web-app-capable" content="yes" />
      <meta name="apple-mobile-web-app-capable" content="yes" />
      <meta name="apple-mobile-web-app-status-bar-style" content="black" />
    </Helmet>
  )
})

export default Head
