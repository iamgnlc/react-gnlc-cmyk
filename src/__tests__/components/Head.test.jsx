import React from 'react';
import Adapter from '@cfaester/enzyme-adapter-react-18';
import { shallow, configure } from 'enzyme';

import Head from '../../components/Head';

configure({ adapter: new Adapter() });

describe('Head', () => {
  beforeEach(() => {
    process.env.REACT_APP_AUTHOR = 'Author';
    process.env.REACT_APP_NAME = 'Name';
  });

  it('should render correctly', () => {
    const wrapper = shallow(<Head />);

    expect(wrapper.getElements()).toMatchSnapshot();
  });
});
