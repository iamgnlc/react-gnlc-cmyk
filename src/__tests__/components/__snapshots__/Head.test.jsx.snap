// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Head should render correctly 1`] = `
Array [
  <HelmetWrapper
    defer={true}
    encodeSpecialCharacters={true}
  >
    <title>
      Author / Name
    </title>
    <link
      href="favicon.ico"
      rel="shortcut icon"
    />
    <meta
      content="noindex"
      name="robots"
    />
    <meta
      charset="utf-8"
    />
    <meta
      content="width=device-width, user-scalable=no, initial-scale=1, minimum-scale=1, maximum-scale=1, minimal-ui"
      name="viewport"
    />
    <meta
      content="Author / Name"
      name="title"
    />
    <meta
      content="Name"
      name="description"
    />
    <meta
      content="Author"
      name="author"
    />
    <meta
      content="Author / Name"
      property="og:site_name"
    />
    <meta
      content="Author / Name"
      property="og:title"
    />
    <meta
      content="Name"
      property="og:description"
    />
    <meta
      content="preview.png"
      property="og:image"
    />
    <meta
      content="Author / Name"
      name="twitter:title"
    />
    <meta
      content="Name"
      name="twitter:description"
    />
    <meta
      content="preview.png"
      name="twitter:image"
    />
    <meta
      content="summary_large_image"
      name="twitter:card"
    />
    <meta
      content="@Author"
      name="twitter:site"
    />
    <meta
      content="Author / Name"
      name="twitter:image:alt"
    />
    <meta
      content="yes"
      name="mobile-web-app-capable"
    />
    <meta
      content="yes"
      name="apple-mobile-web-app-capable"
    />
    <meta
      content="black"
      name="apple-mobile-web-app-status-bar-style"
    />
  </HelmetWrapper>,
]
`;
